app/handler/subjects.go

```diff
@@ -53,15 +53,15 @@ func GetSubject(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
 func UpdateSubject(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 
-	subject := model.Subject{}
+	subject := apiModel.Subject{}
 	err := json.NewDecoder(r.Body).Decode(&subject)
 
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	subject, err = dbOperations.UpdateSubject(cosmos, subject, vars["name"])
+	res, err := dbOperations.UpdateSubject(cosmos, subject, vars["name"])
```

Comment:
Why did you replace subject with res? The response is subject model. Now it is better having storageSubject or apiSubject, but I think it is better having it specified than the generic

Reply:
Put their proper names now. XD